### PR TITLE
Fix issues

### LINKED ISSUE #22: When parent size is controlled by inner text, padding brakes it for smaller texts
https://pixijs.io/layout/storybook/?path=/story/autosize--by-text&args=text:Width;padding:15

### LINKED ISSUE #31: fitInnerText overwrites text wordWrap & wordWrapWidth properties
### Expected Behavior
wordWrap & wordWrapWidth properties, set as style should work
### Current Behavior
they ore overwriten by fitInnerText method
### Possible Solution
add condition and stop overwriting it

### PR DESCRIPTION
**Pull request by CyberDex**
Fixes #22
Fixes #31

### CODE REVIEW

File: src/controllers/SizeController.ts [hunk 1]
```diff
@@ -63,250 +63,205 @@ export class SizeController
...
+                if (needToBeResized)
+                {
+                    this.innerText.style.wordWrap = true;
 
-                    finalWidth = this.innerText.width;
+                    this.innerText.style.wordWrapWidth = availableSpaceHor;
+                }
 
-                    break;
+                const textWidthPaddings = this.innerText.width + paddingLeft + paddingRight;
```
Comment (CyberDex) [inline src/controllers/SizeController.ts:96]: The main change of this PR is here

File: src/controllers/SizeController.ts [hunk 2]
```diff
@@ -63,250 +63,205 @@ export class SizeController
...
-                    }
+                break;
 
-                    break;
-            }
+            case 'static':
+            default:
+                finalWidth = getNumber(width, this.parentWidth);
+                break;
         }
```
Comment (CyberDex) [inline src/controllers/SizeController.ts:167]: Those are just linting as the wrap condition was removed, the code here is the same, it's only `else` block is moved to
```
case 'static':
```

File: src/controllers/SizeController.ts [hunk 3]
```diff
@@ -63,250 +63,205 @@ export class SizeController
...
+        switch (heightModificator)
         {
-            finalWidth = getNumber(width, this.parentWidth);
-        }
+            case 'innerText':
+                // height is auto, there is only 1 child and it is text
+                // resize basing on text height
+                finalHeight = this.innerText?.height + paddingBottom + paddingTop;
 
-        this.fitInnerText(finalWidth);
```
Comment (CyberDex) [inline src/controllers/SizeController.ts:184]: Moved down

File: src/controllers/SizeController.ts [hunk 4]
```diff
@@ -63,250 +63,205 @@ export class SizeController
...
-                    // height is basing on content height
-                    finalHeight = childrenHeight + paddingTop + paddingBottom;
-
-                    break;
-            }
-        }
-        else
-        {
-            finalHeight = getNumber(height, this.parentHeight);
-        }
```
Comment (CyberDex) [inline src/controllers/SizeController.ts:273]: Those are just linting as the wrap condition was removed, the code here is the same, it's only the `else` block that is moved to
```
case 'static':
```

File: src/controllers/SizeController.ts [hunk 5]
```diff
@@ -63,250 +63,205 @@ export class SizeController
...
+                // height is basing on content height
+                finalHeight = childrenHeight + paddingTop + paddingBottom;
 
-            this.fitInnerText(finalWidth);
+                break;
 
-            if (isItJustAText(this.layout) && height === 'auto')
-            {
-                finalHeight = this.innerText?.height + paddingBottom + paddingTop;
-            }
```
Comment (CyberDex) [inline src/controllers/SizeController.ts:295]: No need of this anymore, paddings applied in places where content size is calculated now

File: src/controllers/SizeController.ts [hunk 6]
```diff
@@ -63,250 +63,205 @@ export class SizeController
...
 
-        if (finalWidth === 0 || finalHeight === 0)
-        {
-            this.layout.container.visible = false;
-
-            return;
-        }
-
-        this._width = getNumber(finalWidth, this.parentWidth);
-        this._height = getNumber(finalHeight, this.parentHeight);
```
Comment (CyberDex) [inline src/controllers/SizeController.ts:309]: Moved bit lower

File: src/controllers/SizeController.ts [hunk 7]
```diff
@@ -420,40 +384,70 @@ export class SizeController
...
+            const horScale = width / (textWidth + paddingLeft + paddingRight);
+            const vertScale = height / (textHeight + paddingBottom + paddingTop);
 
-        this.innerText.style.wordWrap = true;
-        this.innerText.style.wordWrapWidth = width - paddingRight - paddingLeft;
+            if (horOverflow || verOverflow)
+            {
+                this.innerText.scale.set(Math.min(horScale, vertScale));
+            }
+        }
```
Comment (CyberDex) [inline src/controllers/SizeController.ts:414]: Second big change of this PR

File: src/controllers/SizeController.ts [hunk 8]
```diff
@@ -420,40 +384,70 @@ export class SizeController
...
 
-        if (background instanceof Container)
+        if (size === 'auto')
         {
-            return 'background';
+            return 'contentSize';
         }
 
-        return 'contentSize';
+        return 'static';
```
Comment (CyberDex) [inline src/controllers/SizeController.ts:450]: Third small but logically important change.

**Review by Zyie — APPROVED** [review]

**Pull request merged**